frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/ProductModal.tsx

```diff
@@ -0,0 +1,285 @@
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  Textarea,
+} from '@chakra-ui/react'
+import { cloneDeep } from 'lodash'
+
+import { Product } from '~shared/types'
+import {
+  centsToDollars,
+  dollarsToCents,
+  formatCurrency,
+} from '~shared/utils/payments'
+
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+import MoneyInput from '~components/MoneyInput'
+import Toggle from '~components/Toggle'
+
+import { useEnv } from '~features/env/queries'
+
+type ProductInput = Product & {
+  display_amount: string
+}
+
+const MIN_QTY_KEY = `min_qty`
+const MAX_QTY_KEY = `max_qty`
+export const ProductModal = ({
+  onClose,
+  onSaveProduct,
+  product, // pass non-null product to edit
+}: {
+  onClose: () => void
+  onSaveProduct: (product: Product) => void
+  product: Product | null
+}) => {
+  const {
+    register,
+    handleSubmit,
+    control,
+    watch,
+    formState: { errors },
+  } = useForm<ProductInput>(
+    product
+      ? {
+          defaultValues: {
+            ...product,
+            display_amount: centsToDollars(product.amount_cents ?? 0),
+          },
+        }
+      : undefined,
+  )
+
+  const watchedInputs = useWatch({
+    control,
+  }) as ProductInput
+
+  // trigger a re-render on children by failing the shallow comparator
+  const clonedWatchedInputs = useMemo(
+    () => cloneDeep(watchedInputs),
+    [watchedInputs],
+  )
+
+  const {
+    data: {
+      maxPaymentAmountCents = Number.MAX_SAFE_INTEGER,
+      minPaymentAmountCents = Number.MIN_SAFE_INTEGER,
+    } = {},
+  } = useEnv()
+
+  const amountValidation: RegisterOptions<ProductInput, 'display_amount'> = {
+    validate: (val) => {
+      // Validate that it is a money value.
+      // Regex allows leading and trailing spaces, max 2dp
+      const validateMoney = /^\s*(\d+)(\.\d{0,2})?\s*$/.test(val ?? '')
+      if (!validateMoney) return 'Please enter a valid payment amount'
+
+      const validateMin = !!val && dollarsToCents(val) >= minPaymentAmountCents
+      // Repeat the check on minPaymentAmountCents for correct typing
+      if (!!minPaymentAmountCents && !validateMin) {
+        return `Please enter a payment amount above ${formatCurrency(
+          Number(centsToDollars(minPaymentAmountCents)),
+        )}`
+      }
+
+      const validateMax = !!val && dollarsToCents(val) <= maxPaymentAmountCents
+      // Repeat the check on maxPaymentAmountCents for correct typing
+      if (!!maxPaymentAmountCents && !validateMax) {
+        return `Please enter a payment amount below ${formatCurrency(
+          Number(centsToDollars(maxPaymentAmountCents)),
+        )}`
+      }
+
+      if (
+        dollarsToCents(val) * clonedWatchedInputs.max_qty >
+        maxPaymentAmountCents
+      ) {
+        return 'Item and Quantity exceeded limit. Either lower your quantity or lower payment amount.'
```

Comment:
grammar seems a bit off here, suggestion: `Total payment amount has exceeded the limit of ___. Either lower the quantity or the payment amount`

(will defer to @staceytan1998 for this)